kpatch-build/create-klp-module.c

```diff
@@ -247,16 +248,17 @@ static void create_klp_arch_sections(struct kpatch_elf *kelf, char *strings)
 	struct rela *rela, *rela2;
 	char *secname, *objname = NULL;
 	char buf[256];
-	int nr, index, offset, old_size, new_size;
+	int nr, index, offset;
+	size_t new_size, old_size;
 
 	karch = find_section_by_name(&kelf->sections, ".kpatch.arch");
 	if (!karch)
 		return;
 
-	nr = karch->data->d_size / sizeof(struct kpatch_arch);
+	nr = (int)(karch->data->d_size / sizeof(struct kpatch_arch));
 
 	for (index = 0; index < nr; index++) {
-		offset = index * sizeof(struct kpatch_arch);
+		offset = (unsigned int)(index * sizeof(struct kpatch_arch));
```

Comment:
This is assigning an `unsigned int` to an `int`, isn't that causing a warning rather than solving one?

Reply:
It should have been `int` instead of `unsigned int`.